### PR TITLE
Feature/mark selected intersections

### PR DESCRIPTION
**Pull request by Outtascope**
Marks selected intersections and waypoints along the solution.l

### CODE REVIEW

File: app/src/main/java/com/example/android/searchabledict/MapDisplay.java [hunk 1]
```diff
@@ -93,6 +93,7 @@
     public static ArrayList<Point> curGPSPoints = null;
 
     public static ArrayList<Point> mIntersectionList = null;
+    public static ArrayList<Point> mWaypointList = null;
```
Comment (Outtascope) [inline app/src/main/java/com/example/android/searchabledict/MapDisplay.java:96]: Storage for the waypoints.

File: app/src/main/java/com/example/android/searchabledict/MapDisplay.java [hunk 2]
```diff
@@ -467,6 +471,7 @@ void updateUI() {
             segment.setEnd(routePoints.get(i));
             path.addSegment(segment, false);  //false so that a new path will not be started
 
+            mWaypointList.add(new Point(routePoints.get(i).getX(), routePoints.get(i).getY()));
```
Comment (Outtascope) [inline app/src/main/java/com/example/android/searchabledict/MapDisplay.java:474]: Collect solution waypoints

File: app/src/main/java/com/example/android/searchabledict/MapDisplay.java [hunk 3]
```diff
@@ -487,6 +492,8 @@ void updateUI() {
             intersectionX = (int) nearbyPoints.get(0).getX();
             intersectionY = (int) nearbyPoints.get(0).getY();
 
+            mIntersectionList.add(new Point(intersectionX, intersectionY));
```
Comment (Outtascope) [inline app/src/main/java/com/example/android/searchabledict/MapDisplay.java:495]: Collect intersection points

File: app/src/main/java/com/example/android/searchabledict/MapDisplay.java [hunk 4]
```diff
@@ -567,6 +574,16 @@ void updateUI() {
...
+        SimpleMarkerSymbol matchedIntersectionSymbol = new SimpleMarkerSymbol(Color.BLUE, 12, SimpleMarkerSymbol.STYLE.CROSS);
+        SimpleMarkerSymbol waypointSymbol = new SimpleMarkerSymbol(Color.GREEN, 8, SimpleMarkerSymbol.STYLE.TRIANGLE);
+
+        for (Point p : mIntersectionList) {
+            intersectionsLayer.addGraphic(new Graphic(p, matchedIntersectionSymbol));
+        }
+
+        for (Point p: mWaypointList) {
+            intersectionsLayer.addGraphic(new Graphic(p, waypointSymbol));b 
+        }
```
Comment (Outtascope) [inline app/src/main/java/com/example/android/searchabledict/MapDisplay.java:586]: Add our intersection and waypoint marks to the graphics layer.

**Review by wagenma9 — APPROVED** [review]

**Pull request merged**